### PR TITLE
Try some support for setting

### PR DESCRIPTION
**Pull request by mickaelistria**
Signed-off-by: Mickael Istria <mistria@redhat.com>

### CODE REVIEW

**Comment by gayanper** [comment]
Any reason we can’t just set the global settings file into mave request object and let maven do the parsing?

File: lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenProjectCache.java [hunk 1]
```diff
@@ -62,16 +61,17 @@
 	private final Map<URI, Collection<ModelProblem>> problemCache;
 	private final PlexusContainer plexusContainer;
 
-	private MavenExecutionRequest mavenRequest;
+	private final MavenExecutionRequest mavenRequest;
 	MavenXpp3Reader mavenReader = new MavenXpp3Reader();
 	private DefaultRepositorySystemSession repositorySystemSession;
 	private ProjectBuilder projectBuilder;
-	private RepositorySystem repositorySystem;
+	//private RepositorySystem repositorySystem;
```
Comment (AObuchow) [inline lemminx-maven/src/main/java/org/eclipse/lemminx/maven/MavenProjectCache.java:68]: Remove

**Comment by mickaelistria** [comment]
> Any reason we can’t just set the global settings file into mave request object and let maven do the parsing?

As far as I can see, the mavenRequest is just a bean. It doesn't contain any logic to parse the settings file.

**Comment by gayanper** [comment]
This change actually fixes the problem of not processing the settings.xml. Once the system properly is set to configuration file everything works like a charm.

**Comment by AObuchow** [comment]
Looks good to me, nice improvement & refactoring @mickaelistria !

**Pull request merged**